index.js

```diff
@@ -126,7 +126,9 @@ function replacePath (path, keys) {
       name: key || index++,
       delimiter: prefix || '/',
       optional: optional,
-      repeat: repeat
+      repeat: repeat,
+      offset: i + (prefix ? prefix.length : 0),
```

Comment:
As post-merge review thought, you might consider replacing some common utility functions with their well-tested lodash versions. e.g. `function omit (keys, src) {` can just be `_.omit`. Since it's just in testing it won't change the weight of the library (or if you're extra concerned you can just import the functions you need since modular lodash exists).